nodebuilder/settings.go

```diff
@@ -105,13 +110,61 @@ func WithMetrics(metricOpts []otlpmetrichttp.Option, nodeType node.Type, buildIn
 	return opts
 }
 
+func WithTraces(opts []otlptracehttp.Option, pyroOpts []otelpyroscope.Option) fx.Option {
+	options := fx.Options(
+		fx.Supply(opts),
+		fx.Supply(pyroOpts),
+		fx.Invoke(initializeTraces),
+	)
+	return options
+}
+
+func initializeTraces(
+	ctx context.Context,
+	nodeType node.Type,
+	peerID peer.ID,
+	network p2p.Network,
+	buildInfo *node.BuildInfo,
+	opts []otlptracehttp.Option,
+	pyroOpts []otelpyroscope.Option,
+) error {
+	var tp trace.TracerProvider
+	client := otlptracehttp.NewClient(opts...)
+	exporter, err := otlptrace.New(ctx, client)
+	if err != nil {
+		return fmt.Errorf("creating OTLP trace exporter: %w", err)
+	}
+
+	tp = tracesdk.NewTracerProvider(
+		tracesdk.WithSampler(tracesdk.AlwaysSample()),
+		// Always be sure to batch in production.
+		tracesdk.WithBatcher(exporter),
+		// Record information about this application in a Resource.
+		tracesdk.WithResource(resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNamespaceKey.String(fmt.Sprintf("Celestia-%s", nodeType)),
+			semconv.ServiceNameKey.String(fmt.Sprintf("semver-%s", buildInfo.SemanticVersion)),
+			semconv.ServiceInstanceIDKey.String(fmt.Sprintf("%s/%s", network.String(), peerID.String()))),
```

Comment:
Need to double check with Devops if this works for them and can be filtered.

Reply:
cc @jrmanes , @Bidon15

Reply:
I can check it, I'm gonna need to build this commit and deploy it to an env, but it's okay, I'll keep you posted guys! thank you!